StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1113CommaMustBeOnSameLineAsPreviousParameter.cs

```diff
@@ -88,361 +88,239 @@ private void HandleArrayDeclaration(SyntaxNodeAnalysisContext context)
-            BaseArgumentListSyntax parameterListSyntax)
+        private static void HandleMethodDeclaration(SyntaxNodeAnalysisContext context)
         {
-            for (int index = 0; index < commas.Count; index++)
-            {
-                var comma = commas[index];
-                if (parameterListSyntax.Arguments.Count <= index)
-                {
-                    return;
-                }
-
-                var previousParameter = parameterListSyntax.Arguments[index];
-
-                var commaLocation = comma.GetLocation();
-                if (commaLocation.GetLineSpan().StartLinePosition.Line !=
-                    previousParameter.GetLocation().GetLineSpan().EndLinePosition.Line)
-                {
-                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, commaLocation));
-                }
-            }
-        }
+            var methodDeclaration = (MethodDeclarationSyntax)context.Node;
+            var parameterList = methodDeclaration.ParameterList;
 
-        private static void CheckIfCommasAreAtTheSameLineAsThePreviousParameter(SyntaxNodeAnalysisContext context, List<SyntaxToken> commas,
-            AttributeArgumentListSyntax attributeListSyntax)
-        {
-            for (int index = 0; index < commas.Count; index++)
+            if (parameterList != null && !parameterList.IsMissing)
             {
-                var comma = commas[index];
-                if (attributeListSyntax.Arguments.Count <= index)
-                {
-                    return;
-                }
-
-                var previousParameter = attributeListSyntax.Arguments[index];
-
-                var commaLocation = comma.GetLocation();
-                if (commaLocation.GetLineSpan().StartLinePosition.Line !=
-                    previousParameter.GetLocation().GetLineSpan().EndLinePosition.Line)
+                var parameters = parameterList.Parameters;
+                if (parameters.Count > 1)
                 {
-                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, commaLocation));
+                    CheckIfCommasAreAtTheSameLineAsThePreviousParameter(context, parameters.GetWithSeparators());
                 }
             }
         }
 
-        private static void CheckIfCommasAreAtTheSameLineAsThePreviousParameter(SyntaxNodeAnalysisContext context, List<SyntaxToken> commas,
-            AttributeListSyntax attributeListSyntax)
+        private static void CheckIfCommasAreAtTheSameLineAsThePreviousParameter(SyntaxNodeAnalysisContext context, SyntaxNodeOrTokenList nodeOrTokenList)
         {
-            for (int index = 0; index < commas.Count; index++)
+            // If flag is even we expecting parameter syntax node, otherwise we expecting comma token.
+            int index = 0;
+            SyntaxNode previousNode = null;
+
+            foreach (var nodeOrToken in nodeOrTokenList)
```

Comment:
As an index is needed, it would probably be better to use a `for` loop here, instead of `foreach`. Especially since `SyntaxNodeOrTokenList` supports an indexer